File: code/1040/1040.rb
inputs = gets.split(' ')

n1 = inputs[0].to_f * 2
n2 = inputs[1].to_f * 3
n3 = inputs[2].to_f * 4
n4 = inputs[3].to_f

average = (n1 + n2 + n3 + n4) / 10

puts format('Media: %.1f', average)

if average < 5
  puts 'Aluno reprovado.'
elsif average > 7
  puts 'Aluno aprovado.'
else
  puts 'Aluno em exame.'
  exam = gets.to_f
  puts format('Nota do exame: %.1f', exam)
  average_f = (average + exam) / 2
  if average_f < 5
    puts 'Aluno reprovado.'
  else
    puts 'Aluno aprovado.'
  end
  puts format('Media final: %.1f', average_f)
end

File: code/1040/1040.c
#include <stdio.h>

int main() {

  double n1, n2, n3, n4, exam, average, averageF;

  scanf("%lf %lf %lf %lf", &n1, &n2, &n3, &n4);

  average = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10;

  printf("Media: %.1lf\n", average);

  if (average < 5) {
    printf("Aluno reprovado.\n");
    return 0;
  }

  if (average > 7) {
    printf("Aluno aprovado.\n");
    return 0;
  }

  printf("Aluno em exame.\n");

  scanf("%lf", &exam);

  printf("Nota do exame: %.1lf\n", exam);

  averageF = (average + exam) / 2;

  if (averageF < 5) {
    printf("Aluno reprovado.\n");
  } else {
    printf("Aluno aprovado.\n");
  }

  printf("Media final: %.1lf\n", averageF);

  return 0;
}
